Source/Core/VideoBackends/OGL/TextureConverter.cpp

```diff
@@ -269,18 +269,27 @@ static void EncodeToRamUsingShader(GLuint srcTexture,
 	}
 }
 
-void EncodeToRamFromTexture(u8 *dest_ptr, const TextureCache::TCacheEntryBase *texture_entry,
-                            GLuint source_texture, bool bFromZBuffer, bool bIsIntensityFmt, int bScaleByHalf, const EFBRectangle& source)
+void EncodeToRamFromTexture(u8 *dest_ptr, u32 format, u32 native_width, u32 bytes_per_row, u32 num_blocks_y, u32 memory_stride,
+                            PEControl::PixelFormat srcFormat, bool bIsIntensityFmt, int bScaleByHalf, const EFBRectangle& source)
 {
-	SHADER& texconv_shader = GetOrCreateEncodingShader(texture_entry->format);
+	g_renderer->ResetAPIState();
+
+	SHADER& texconv_shader = GetOrCreateEncodingShader(format);
```

Comment:
This will call ResetAPIState twice if efb2ram and efb2tex is used.
@JMC47 So did you see a slowdown compared to master when both are enabled?